src/styles/index.scss

```diff
@@ -1,3 +1,56 @@
+@import '/src/styles/variables';
+
 body {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $with;
+  gap: $size-gap;
+
+  &_day {
+    display: flex;
+    height: $size-day;
+    width: $size-day;
+    box-sizing: border-box;
+    align-items: center;
+    justify-content: center;
+    background-color: #eee;
+    border: 1px solid black;
+    transition: all 500ms;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--month-length-#{$i} .calendar_day:nth-child(n + #{$i + 1}) {
+    display: none;
+  }
+}
+
+@each $name-day, $nomber-dey in $day {
+  .calendar--start-day-#{$name-day} {
+    .calendar_day:first-child {
```

Comment:
There is a typo in the class name `.calendar_day`. It should be `.calendar__day`. According to the BEM methodology, element names are separated by two underscores. It's important to follow the BEM methodology for better code readability and maintainability.